### ISSUE TITLE
Homebrew formula from #3 writes "unexpected files" (according to `brew doctor`)

### ISSUE THREAD

**Issue by AlexanderWillner** [macos-fuse-t/fuse-t#36]
Description: Homebrew formula from #3 writes "unexpected files" (according to `brew doctor`). These are just warnings, the locations are fine (for brewed formulae), but 
Version: `1.0.27` (fuse-t) / `13.5.1` (macOS on M1/M2)

**Steps that will reproduce the problem?**
1. Run `brew reinstall macos-fuse-t/cask/fuse-t`
2. Run `brew doctor`

**What is the expected result?**
`Your system is ready to brew.`

**What happens instead?**
See below.

**Possible workaround:**
Ignore it, it's just a warning. It seems that non-cask formulae write to `/usr/local/Cellar/NAME/VERSION/lib` and then some / few link these libraries to `/usr/local/lib/`.

**Any additional information:**
 

```shell
% brew doctor
Your system is ready to brew.
% brew reinstall macos-fuse-t/cask/fuse-t
...
% brew doctor                            
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libfuse-t-1.0.27.dylib

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/fuse-t.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libfuse-t-1.0.27.a
```

**Comment by vtwigt** [comment]
Still applies to Fuse-T 1.0.38:

```shell
% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libfuse-t-1.0.38.dylib

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/fuse-t.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libfuse-t-1.0.38.a
```

**Comment by makorus** [comment]
Same here on my device, so this must be an issue with the cask itself. Would love to see this one fixed soon.

**Comment by nicfab** [comment]
Same here. I hope it will be fixed soon.

**Comment by alexfs** [comment]
This is not a problem, you should ignore those warnings.

**Comment by darrenoh** [comment]
I am trying to create a formula for ext4fuse, and Homebrew cannot find the fuse-t library. I suspect this issue is the reason.

**Comment by gromgit** [comment]
> I am trying to create a formula for ext4fuse, and Homebrew cannot find the fuse-t library. I suspect this issue is the reason.

No, this issue is about `brew doctor` warning about stuff in `/usr/local` that ARM Homebrew definitely couldn't have installed (all ARM formulae are confined to `/opt/homebrew`), but may affect the Homebrew formulae you installed.

However, what you're experiencing is related: Homebrew actively _removes_ all references to stuff under `/usr/local` during ARM source builds, precisely to avoid the above situation (hidden dependencies that cause installs on other machines to fail mysteriously). See any of the formulae under [my FUSE tap](https://github.com/gromgit/homebrew-fuse) for a way around this restriction that's specific to MacFUSE. It may or may not work unchanged for fuse-t formulae, but it shouldn't be hard to figure out.

**Comment by justauserid** [comment]
> If you used the release .pkg installer, then I _think_ you need to `sudo /Library/Application Support/fuse-t/uninstall.sh`.

Just a tangential _guess_ - this could be related to #83. (I am saying "guess" - because I didn't isntall with package installer).

Because two (apparently) most famous fuse-t guides (one at the project github itself and another on Cryptomator docs) instructs this:

```
brew tap macos-fuse-t/homebrew-cask
brew install fuse-t
brew install fuse-t-sshfs
```

But someone has added `fuse-t` to core tap https://formulae.brew.sh/cask/fuse#default but `fuse-t-sshfs` is not added to the core tap. So one ends up isntalling half of the requirement from core tap and another half from project tap.

So could this be the problem?

PS. And I also do not understand – when I added `macos-fuse-t/homebrew-cask` as the tap, then how does it show up as `macos-fuse-t/cask` when I list them with `brew tap`? Where does the "homebrew-" part of the cask name after / go? :-)

**Comment by justauserid** [comment]
When I tried to replicate it - installed from both sources and then removed the .pkg installed one (with the sudo script call you had listed above and it ran successfully) and then I tried to remove the brew one as well with `brew remove fuse-t` but it expects that uninstaller script (apparently) from .pkg installer:

```
% brew remove fuse-t                           
==> Uninstalling Cask fuse-t
==> Running uninstall script /Library/Application Support/fuse-t/uninstall.sh
Error: uninstall script /Library/Application Support/fuse-t/uninstall.sh does not exist.
```

Or is it expected behaviour?

**Comment by justauserid** [comment]
I even tried `brew install macos-fuse-t/homebrew-cask/fuse-t` (before removing `fuse-t` - which must have been from core tap I guess) and still the same `brew doctor` warning..

**Comment by aj0413** [comment]
@alepee just noticed this today as I'm doing some system spring cleaning

Can we alter the auth to set to current user instead of root?

**Comment by AlexanderWillner** [comment]
lgtm, thanks!

**Comment by matthewjmiller1** [comment]
@gromgit does the brew `allow_list` need updated again to include the below static library?

```
Mac$ brew doctor; brew cleanup
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libfuse3.a
Mac$ brew --version
Homebrew 5.1.7
Mac$
```

**Comment by gromgit** [comment]
Hi @macos-fuse-t, can you confirm this static lib was added recently? Thanks much!

**Comment by macos-fuse-t** [comment]
correct

**Comment by gromgit** [comment]
> [@gromgit](https://github.com/gromgit) does the brew `allow_list` need updated again to include the below static library?

It does now. `brew update` and try again.
